### PR TITLE
[3.1][Security] Fix DebugAccessDecisionManager when object is not a scalar

### PR DESCRIPTION
**Pull request by romainneutron**
| Q | A |
| --- | --- |
| Branch? | 3.1 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | N/A |
| License | MIT |

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by romainneutron** [comment]
I've not added test because this class is not tested at all. If you want, I can add some

**Comment by stof** [comment]
would be great to add some tests, yeah

File: src/Symfony/Component/Security/Core/Authorization/DebugAccessDecisionManager.php [hunk 1]
```diff
@@ -103,7 +103,11 @@ private function getStringRepresentation($object)
         }
 
         if (!is_object($object)) {
-            return sprintf('%s (%s)', gettype($object), $object);
+            if (is_scalar($object)) {
+                return sprintf('%s (%s)', gettype($object), $object);
+            } else {
```
Comment (stof) [inline src/Symfony/Component/Security/Core/Authorization/DebugAccessDecisionManager.php]: ``else`` is useless

**Comment by romainneutron** [comment]
PR updated, tests added

File: src/Symfony/Component/Security/Core/Tests/Authorization/DebugAccessDecisionManagerTest.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        $adm->decide($this->getMock(TokenInterface::class), ['ATTRIBUTE_1'], $object);
+
+        $this->assertSame($expectedLog, $adm->getDecisionLog());
+    }
+
+    public function provideObjectsAndLogs()
+    {
+        $object = new \stdClass();
+
+        yield [[['attributes' => ['ATTRIBUTE_1'], 'object' => 'NULL', 'result' => false]], null];
```
Comment (stof) [inline src/Symfony/Component/Security/Core/Tests/Authorization/DebugAccessDecisionManagerTest.php]: we don't use the short array syntax in Symfony

File: src/Symfony/Component/Security/Core/Authorization/DebugAccessDecisionManager.php [hunk 3]
```diff
@@ -103,7 +103,13 @@ private function getStringRepresentation($object)
         }
 
         if (!is_object($object)) {
-            return sprintf('%s (%s)', gettype($object), $object);
+            if (is_bool($object)) {
+                return sprintf('%s (%s)', gettype($object), $object ? 'true' : 'false');
+            } elseif (is_scalar($object)) {
```
Comment (stof) [inline src/Symfony/Component/Security/Core/Authorization/DebugAccessDecisionManager.php]: should simply be ``if``

**Comment by romainneutron** [comment]
PR updated, comments addressed

**Comment by fabpot** [comment]
Thank you @romainneutron.

**Pull request merged**